python/src/aiconfig/editor/server/server.py

```diff
@@ -566,3 +567,48 @@ def _op(aiconfig_runtime: AIConfigRuntime, _op_args: OpArgs) -> Result[None, str
 
     signature: dict[str, Type[Any]] = {}
     return run_aiconfig_operation_with_request_json(aiconfig, request_json, f"method_", _op, signature)
+
+
+@app.route("/api/get_cli_config", methods=["GET"])
+def get_cli_config() -> FlaskResponse:
+    state = get_server_state(app)
+
+    def _yaml_deserialize(yaml_str: str) -> Result[dict[str, Any], str]:
+        try:
+            out = yaml.safe_load(yaml_str)
+            if out is None:
+                return Err("YAML string not parsed")
```

Comment:
I know we chain this, but can since `_yaml_deserialize` is only used it one place, can we call this something like `_parse_yaml_file(filename: str)` and return a more user friendly error?

`Err(f"Cannot parse string from YAML file '{filename}' into YAML format")